modules/imgproc/src/cornersubpix.cpp

```diff
@@ -96,7 +96,7 @@ void cv::cornerSubPix( InputArray _image, InputOutputArray _corners,
     for( int pt_i = 0; pt_i < count; pt_i++ )
     {
         Point2f cT = corners[pt_i], cI = cT;
-        CV_Assert( Rect(0, 0, src.cols, src.rows).contains(cT) );
+        CV_Assert( Rect2f(0.0f, 0.0f, static_cast<float>(src.cols), static_cast<float>(src.rows)).contains(cT) );
```

Comment:
Check line 145 too.

Please note that conversion int->float may lead to accuracy errors.

May be we should try to fix `contains` method instead?

Reply:
Thank you for your comment !

> May be we should try to fix contains method instead?

I believe this will be a better direction, so I tried implementing it.

- Point2f converts to Point2d with Rect_<int>.
- Rect_<int> converts to Rect_<double> only with Point2d.

Reply:
Because keyPoint has float type, various side effects occur(features2d/stiching/...). 

I'll take a bit more time to fix it.

Reply:
I tried to fix it. And this is breaking change, so I add warning on this function.

https://pullrequest.opencv.org/buildbot/export/pr/26022/docs/d2/d44/classcv_1_1Rect__.html#a4506eca68c36d1c68af51489a7ac58a7